lib/sdam/topology.js

```diff
@@ -149,6 +150,28 @@ class Topology extends EventEmitter {
 
     // add client info
     this.s.clientInfo = createClientInfo(options);
+
+    if (options.srvHost) {
```

Comment:
I do like that idea, but I don't think I'd want to implement it now:

+ It's nice that we don't have to worry about options changing after they are passed into the topology. TXT records with extra options especially come to mind
+ It would be a bigger deviation in our connection string parsing.